### PR TITLE
feat: Add `microphone_connect` action

### PR DESCRIPTION
**Pull request by felipecrs**
~To be merged after #1536.~

### CODE REVIEW

**Comment by dermotduffy** [comment]
I'm not sure if this will work in the normal case, as connecting the microphone is obviously a highly sensitive action (unlike disconnecting). I am guessing the browser would reject this?

**Comment by felipecrs** [comment]
> I'm not sure if this will work in the normal case, as connecting the microphone is obviously a highly sensitive action (unlike disconnecting). I am guessing the browser would reject this?

Hm... for what I was wondering, it is working fine. I used it in the green call button demonstrated in: https://github.com/dermotduffy/frigate-hass-card/pull/1536#issuecomment-2351682720

Maybe however it would not work as a URL query parameter (I did not test).

**Comment by dermotduffy** [comment]
> Hm... for what I was wondering, it is working fine. I used it in the green call button demonstrated in: https://github.com/dermotduffy/frigate-hass-card/pull/1536#issuecomment-2351682720

Actual, on thinking about it, worst case the browser would just use ask the user as usual. 

Ready for me to look at this one?

**Comment by felipecrs** [comment]
> Actual, on thinking about it, worst case the browser would just use ask the user as usual.

That's correct. Anyway, this is certainly less sensitive than `microphone_unmute` (which causes `connect` as well), which is already implemented and work well.[](url)

> Ready for me to look at this one?

Yes, please. Very straightforward implementation thanks to your [instructions](https://github.com/dermotduffy/frigate-hass-card/issues/1457#issuecomment-2345258607).

File: src/card-controller/actions/actions/microphone-connect.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { GeneralActionConfig } from '../../../config/types';
+import { CardActionsAPI } from '../../types';
+import { FrigateCardAction } from './base';
+
+export class MicrophoneConnectAction extends FrigateCardAction<GeneralActionConfig> {
+  public async execute(api: CardActionsAPI): Promise<void> {
+    api.getMicrophoneManager().connect();
```
Comment (dermotduffy) [inline src/card-controller/actions/actions/microphone-connect.ts:7]:
```suggestion
    await api.getMicrophoneManager().connect();
```

Comment (felipecrs) [reply src/card-controller/actions/actions/microphone-connect.ts]: Good catch!

**Comment by dermotduffy** [comment]
Thanks @felipecrs !

**Pull request merged**